src/prog_models/prognostics_model.py

```diff
@@ -1085,4 +1087,344 @@ def optimization_fcn(params):
+
+            for iter_data in range(len(self.states)):
+                state_name = self.states[iter_data]
+                states_data_temp = [states[iter_data1][state_name] for iter_data1 in range(len(states))]
+                states_data_interp[state_name] = interp1d(times,states_data_temp)(time_data_interp)
+            for iter_data3 in range(len(self.inputs)):
+                input_name = self.inputs[iter_data3]
+                inputs_data_temp = [inputs[iter_data4][input_name] for iter_data4 in range(len(inputs))]
+                inputs_data_interp[input_name] = interp1d(times,inputs_data_temp)(time_data_interp)
+
+            states_data = []
+            inputs_data = [] 
+            for iter_dataT in range(len(time_data_interp)):
+                states_data.append(self.StateContainer({key: states_data_interp[key][iter_dataT] for key in states_data_interp.keys()}))
+                inputs_data.append(self.InputContainer({key: inputs_data_interp[key][iter_dataT] for key in inputs_data_interp.keys()}))
+
+            times = time_data_interp.tolist()
+            states = SimResult(time_data_interp,states_data)
+            inputs = SimResult(time_data_interp,inputs_data)
+            outputs = LazySimResult(self.output, time_data_interp, states_data) 
+            event_states = LazySimResult(self.event_state, time_data_interp, states_data)
+
+            # Only save values that the user designated to be included in surrogate model
+            if len(config['states_dmd']) is not len(self.states):
+                for key in self.states:
+                    if key not in config['states_dmd']:
+                        if iter_load == 0:
+                            states_dmd.remove(key)
+                        for iter in range(len(times)):
+                            del states[iter][key]
+            if len(config['inputs_dmd']) is not len(self.inputs):
+                for key in self.inputs:
+                    if key not in config['inputs_dmd']:
+                        if iter_load == 0:
+                            inputs_dmd.remove(key)
+                        for iter in range(len(times)):
+                            del inputs[iter][key]
+            if len(config['outputs_dmd']) is not len(self.outputs):
+                for key in self.outputs:
+                    if key not in config['outputs_dmd']:
+                        if iter_load == 0:
+                            outputs_dmd.remove(key)
+                        for iter in range(len(times)):
+                            del outputs[iter][key] 
+            if len(config['events_dmd']) is not len(self.events):
+                for key in self.events:
+                    if key not in config['events_dmd']:
+                        if iter_load == 0:  
+                            events_dmd.remove(key)
+                        for iter in range(len(times)):
+                            del event_states[iter][key]      
```

Comment:
Maybe we could put this into a sub-function? The differing elements are keys iterated through, key in config[], object to remove key from, and object to delete[iter][key] from. What if this was passed into a function to perform the repeated code, compressed into something like this:
```suggestion
            # need to make sure these are passed by reference (python does by default?) 
            def user_val_set(iter_loop : list, config_key : str, remove_from : dict, del_from) -> None:
            """Sub-function for performing check and removal for user designated values.
            
            Args:
                iter_loop : list
                    List of keys to iterate through.
                config_key : str
                    String key to check keys against config
                remove_from : dict
                    Dictionary dmd to remove key from
                del_from : list or dict?
                    Final data structure we call del ...[iter][key] on
            """
               for key in iter_loop:
                   if key not in config[config_key]:
                       if iter_load == 0:
                           remove_from.remove(key)
                       for iter in range(len(times)):
                           del del_from[iter][key]
                           
            if len(config['states_dmd']) is not len(self.states):
                user_val_set(self.states,  'states_dmd', states_dmd, states)
            if len(config['inputs_dmd']) is not len(self.inputs):
                user_val_set(self.inputs,  'inputs_dmd', inputs_dmd, inputs)
            if len(config['outputs_dmd']) is not len(self.outputs):
                user_val_set(self.outputs,  'outputs_dmd', outputs_dmd, outputs) 
            if len(config['events_dmd']) is not len(self.events):
                user_val_set(self.events,  'events_dmd', events_dmd, event_states)      
```

Reply:
Thanks for the suggestion, this is much cleaner